lib/kubernetes-deploy/deploy_task.rb

```diff
@@ -86,7 +86,7 @@ def prune_whitelist
         policy/v1beta1/PodDisruptionBudget
         batch/v1beta1/CronJob
       )
-      wl + cluster_resource_discoverer.crds(@sync_mediator).select(&:prunable?).map(&:group_version_kind)
+      wl + cluster_resource_discoverer.crds.select(&:prunable?).map(&:group_version_kind)
```

Comment:
I can't put my finger on why, but it seems odd to me that ResourceDiscovery has its own cache. Do you have a guiding light for what should share a cache and what shouldn't?

Reply:
Everything involved in a sync cycle of some kind (so checking initial resource statuses or the watcher) should share a cache. ResourceDiscovery's cache is valid for the lifetime of the instance (and of the deploy really) and there's nothing else that shares this lifecycle. It doesn't really need to use a ResourceCache at all. Maybe it would be better to give it a `Kubeclient`.